ipywidgets_jsonschema/form.py

```diff
@@ -197,6 +211,21 @@ def _construct(self, schema, label=None, root=False):
             raise FormError("Expecting type information for non-enum properties")
         if not isinstance(type_, str):
             raise FormError("Not accepting arrays of types currently")
+
+        format_ = schema.get("format", None)
+        available_fmt = []
+        if hasattr(ipywidgets, "DatetimePicker"):
+            available_fmt.append("date-time")
+        if hasattr(ipywidgets, "DatePicker"):
+            available_fmt.append("date")
+        if hasattr(ipywidgets, "TimePicker"):
+            available_fmt.append("time")
```

Comment:
What is the rationale behind these checks? Have the relevant widgets been introduced very recently into `ipywidgets`?

Reply:
Yes, they were added in 8. The test suite also runs agains 7, so in order to pass the tests I had to add these checks.

Reply:
Ah yes, I have slightly rewritten this in #87, as I would like to avoid doing to much `hasattr` stuff.